BrainPortal/app/models/user.rb

```diff
@@ -423,7 +427,8 @@ def union_group_ids_from_access_profiles
   # group 99, because it's present in ap2.
   def apply_access_profiles(remove_group_ids: [])
     gids = union_group_ids_from_access_profiles
-    self.group_ids = self.group_ids - remove_group_ids + gids # - and + are NOT COMMUTATIVE!
+    self.group_ids = self.group_ids - remove_group_ids + gids.to_set.to_a # - and + are NOT COMMUTATIVE!
```

Comment:
Darn, I was sure Rails would remove duplicates; so the bug fix (to be clean) should go at both places.
Here,

```ruby
self.group_ids = (self.group_ids - remove_group_ids + gids).uniq
```